src/router.rs

```diff
@@ -2,11 +2,18 @@ use crate::{
     config::Configuration,
     routes::{health_check, zome_call},
     service::AppState,
+    ReconnectingAdminWebsocket,
 };
 use axum::{http::StatusCode, routing::get, Router};
 
 pub fn hc_http_gateway_router(configuration: Configuration) -> Router {
-    let state = AppState { configuration };
+    let admin_ws = ReconnectingAdminWebsocket::new(configuration.admin_ws_url.as_ref());
```

Comment:
This will need to be a parameter to `hc_http_gateway_router` as a trait. See https://github.com/holochain/hc-http-gw/pull/23

Otherwise we can't do unit testing that doesn't connect to Holochain